packages/ui-components/src/components/utils.ts

```diff
@@ -12,3 +12,23 @@ export const getConfig = (): UIComponentsConfig => {
 
 	return DEFAULT_CONFIG;
 };
+
+export const getCollapsedElement = (elements: ChildNode[]): HTMLElement | undefined => {
+	if (elements.length === 0) {
+		return;
+	}
+
+	const [firstElement, ...siblings] = elements as HTMLElement[];
+
+	if (isOverflowEllipsis(firstElement as HTMLElement)) {
+		return firstElement;
+	}
+
+	return getCollapsedElement(Array.from(firstElement.childNodes)) ?? getCollapsedElement(siblings);
+};
+
+export const isOverflowEllipsis = (element: HTMLElement): boolean => isNodeText(element) && getCssStyle(element, 'text-overflow') === 'ellipsis';
+
+export const isNodeText = (element: HTMLElement): boolean => element.nodeName !== '#text';
```

Comment:
Node is a Text node if nodeName is different from `#text`??

Reply:
Nope, that's wrong. I'm going to change it, nice catch :muscle: